### PR TITLE
Issue 66 - Remove Grok

### LINKED ISSUE #66: remove grok
'nuf said.

### PR DESCRIPTION
**Pull request by rodfersou**
closes #66

### CODE REVIEW

File: src/collective/upload/viewlets.zcml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    xmlns:five="http://namespaces.zope.org/five"
+    xmlns:i18n="http://namespaces.zope.org/i18n"
+    i18n_domain="collective.upload">
+
+    <browser:viewlet
+        for="*"
```
Comment (hvelarde) [inline src/collective/upload/viewlets.zcml]: I think this should be available only to folderish objects.

File: src/collective/upload/browser.zcml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    <browser:page
+        name="upload-settings"
+        for="Products.CMFPlone.interfaces.IPloneSiteRoot"
+        class=".controlpanel.UploadSettingsControlPanel"
+        permission="collective.upload.Setup"
+        layer=".interfaces.IUploadBrowserLayer"
+        />
+
+    <browser:page
+        for="*"
```
Comment (hvelarde) [inline src/collective/upload/browser.zcml]: I think this should be available only to folderish objects.

File: src/collective/upload/browser.zcml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        name="media_uploader"
+        class=".browser.MediaUploader"
+        permission="collective.upload.UploadFiles"
+        template="templates/media_uploader.pt"
+        layer=".interfaces.IUploadBrowserLayer"
+        />
+
+    <browser:page
+        for="*"
```
Comment (hvelarde) [inline src/collective/upload/browser.zcml]: I think this should be available only to folderish objects.

File: src/collective/upload/browser.zcml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        name="api"
+        class=".browser.JSONView"
+        permission="cmf.AddPortalContent"
+        layer=".interfaces.IUploadBrowserLayer"
+        />
+
+    <browser:page
+        for="*"
```
Comment (hvelarde) [inline src/collective/upload/browser.zcml]: I think this should be available only to folderish objects.

File: src/collective/upload/browser.zcml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        name="jsvariables"
+        class=".browser.JSVariables"
+        permission="zope.Public"
+        layer=".interfaces.IUploadBrowserLayer"
+        />
+
+    <browser:page
+        for="*"
```
Comment (hvelarde) [inline src/collective/upload/browser.zcml]: I think this should be available only to folderish objects.

File: src/collective/upload/browser.zcml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        name="jsonimageserializer"
+        class=".browser.JSONImageConverter"
+        permission="zope.Public"
+        layer=".interfaces.IUploadBrowserLayer"
+        />
+
+    <browser:page
+        for="*"
```
Comment (hvelarde) [inline src/collective/upload/browser.zcml]: I think this should be available only to folderish objects.

**Comment by hvelarde** [comment]
@petri can you help us reviewing this?

**Comment by petri** [comment]
Seems good to me. +1 on narrowing down the registrations as suggested by @hvelarde.

**Pull request merged**